### PR TITLE
Disallow global jquery and jest

### PR DESCRIPTION
**Pull request by mikenikles**
Removes the ESLint `jquery` and `jest` environments globally and instead relies on explicit configuration in files where global variables of these environments are necessary.

The main benefit of this PR is to enforce the use of our internal `$` variable returned by the SDK's `jqueryResponse` function, rather than the global `$` bound to `window` in browsers.

## Changes
- Removed the ESLint env settings.
- Updated all test files accordingly.

## How to test-drive this PR
- Run `npm run test:js`.
  - Verify that it succeeds.
- Open any file and try to use `$`.
  - Verify that ESLint reports an error.
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)

### CODE REVIEW

File: web/app/containers/login/parsers/common.js [hunk 1]
```diff
@@ -29,3 +29,5 @@ export const parseHiddenInputs = ($, $inputs) => {
         }
     })
 }
+
+export const isFormResponseInvalid = ($, $response, formSelector) => $response.find(formSelector).length
```
Comment (jnurse) [inline web/app/containers/login/parsers/common.js]: It looks like `$` isn't needed here.

File: web/app/containers/login/actions.js [hunk 2]
```diff
@@ -94,12 +96,10 @@ const validateRegisterForm = (formValues) => {
...
-        .then((response) => {
-            return response.text()
-        })
-        .then((responseText) => {
-            const $html = $(responseText)
-            if ($html.find(formSelector).length) {
+        .then(jqueryResponse)
+        .then((res) => {
+            const [$, $response] = res // eslint-disable-line no-unused-vars
+            if (isFormResponseInvalid($response)) {
```
Comment (jnurse) [inline web/app/containers/login/actions.js]: Shouldn't the formSelector be passed to this function too?

**Comment by mikenikles** [comment]
@jnurse Let's never talk about the code I originally pushed... :). I addressed your comments in 37bbc9b

**Comment by jnurse** [comment]
Looks good! 👍

**Pull request merged**